### PR TITLE
Fix accessor assignment

### PR DESCRIPTION
**Pull request by TheLartians**
Enables using accessors as expressions.

### CODE REVIEW

File: src/LuaTransformer.ts [hunk 1]
```diff
@@ -1797,7 +1797,10 @@ export class LuaTransformer {
             );
         }
 
-        if (ts.isPropertyAccessExpression(expression.left) || ts.isElementAccessExpression(expression.left)) {
+        if (
+            ts.isPropertyAccessExpression(expression.left) && !tsHelper.hasSetAccessor(expression.left, this.checker)
```
Comment (TheLartians) [inline src/LuaTransformer.ts]: probably should also check for get accessor

File: src/LuaTransformer.ts [hunk 2]
```diff
@@ -1797,7 +1797,10 @@ export class LuaTransformer {
             );
         }
 
-        if (ts.isPropertyAccessExpression(expression.left) || ts.isElementAccessExpression(expression.left)) {
+        if (
```
Comment (Perryvw) [inline src/LuaTransformer.ts]: This is a strange line break

Comment (Perryvw) [reply src/LuaTransformer.ts]: oh nvm I see how it's laid out now, maybe add parentheses around the && statement to indicate order though

Comment (TheLartians) [reply src/LuaTransformer.ts]: ok I added parentheses

**Pull request merged**